src/com/facebook/buck/features/go/CGoGenSource.java

```diff
@@ -75,8 +75,7 @@ public CGoGenSource(
     for (SourcePath srcPath : cgoSrcs) {
       String path =
           projectFilesystem
-              .getRootPath()
-              .toAbsolutePath()
+              .getPathForRelativePath(getBuildTarget().getBasePath())
```

Comment:
would it make sense to provide a backwards compatibility for users using an older version of cgo?

Reply:
@monty-uber is working on a big change that has a side effect of adding the backward compatibility. The change is not self contained, and cannot be cherry-picked to this PR.

Before his PR is ready, this PR will enable Buck work with the latest version of Go.